src/main/java/org/apache/commons/compress/compressors/lha/AbstractLhStaticHuffmanCompressorInputStream.java

```diff
@@ -0,0 +1,344 @@
+
+                if (index == 2) {
+                    // After reading the first three code lengths, we read a 2-bit skip range
+                    index += (int) bin.readBits(2);
+                }
+            }
+
+            return new BinaryTree(codeLengths);
+        }
+    }
+
+    /**
+     * Read code length (depth in tree). Usually 0-7 but could be higher and if so,
+     * count the number of following consecutive one bits and add to the length.
+     *
+     * @return code length
+     * @throws IOException if an I/O error occurs
+     */
+    protected int readCodeLength() throws IOException {
+        int len = (int) bin.readBits(CODE_LENGTH_BITS);
+        if (len == 0x07) {
+            // Count the number of following consecutive one bits
+            while (bin.readBit() == 1) {
+                if (len == MAX_CODE_LENGTH) {
+                    throw new CompressorException("Code length overflow");
+                }
+
+                len++;
+            }
+        }
+
+        return len;
+    }
+
+    /**
+     * Read the command tree which is used to decode the commands (literals or copy commands).
+     *
+     * @param commandDecodingTree the Huffman tree used to decode the command lengths
+     * @return the command tree
+     * @throws IOException if an I/O error occurs
+     */
+    protected BinaryTree readCommandTree(final BinaryTree commandDecodingTree) throws IOException {
+        final int numCodeLengths = (int) bin.readBits(COMMAND_TREE_LENGTH_BITS);
+
+        if (numCodeLengths > getMaxNumberOfCommands()) {
+            throw new CompressorException("Code length table has invalid size (%d > %d)", numCodeLengths, getMaxNumberOfCommands());
+        } else if (numCodeLengths == 0) {
+            // If numCodeLengths is zero, we read a single code length of COMMAND_TREE_LENGTH_BITS bits and use as root of the tree
+            return new BinaryTree(new int[] { (int) bin.readBits(COMMAND_TREE_LENGTH_BITS) });
+        } else {
+            // Read all code lengths
+            final int[] codeLengths = new int[numCodeLengths];
+
+            for (int index = 0; index < numCodeLengths;) {
+                final int codeOrSkipRange = commandDecodingTree.read(bin);
+
+                if (codeOrSkipRange == 0) {
+                    // Skip one code length
+                    index++;
+                } else if (codeOrSkipRange == 1) {
+                    // Skip a range of code lengths, read 4 bits to determine how many to skip
+                    index += (int) bin.readBits(4) + 3;
+                } else if (codeOrSkipRange == 2) {
+                    // Skip a range of code lengths, read 9 bits to determine how many to skip
+                    index += (int) bin.readBits(9) + 20;
+                } else {
+                    // Subtract 2 from the codeOrSkipRange to get the code length
+                    codeLengths[index++] = codeOrSkipRange - 2;
```

Comment:
I don't think you want to be here if `commandDecodingTree.read(bin)` returns -1.

Reply:
I do check for EOF when `codeOrSkipRange` returns -1 above? What am I missing?